pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -406,9 +414,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env *model.En
 			}
 
 			bind := ingressListener.Bind
-			// if bindToPort is true, we set the bind address if empty to 127.0.0.1
+			// if bindToPort is true, we set the bind address if empty to 0.0.0.0 - this is an inbound port.
 			if len(bind) == 0 && bindToPort {
-				bind = LocalhostAddress
+				bind = WildcardAddress
```

Comment:
are you sure you want to start listening on all the interfaces? Its kind of dangerous. Its easier for the user to specify 0.0.0.0 explicitly

Reply:
There is no other choice - this is the inbound listener. That's what we do for REDIRECT - accept traffic from any host. 

User can override 0.0.0.0 explicitly - but the default can't be useless, and I don't know any case your target port will not be usable from outside.